test-system/src/lib.rs

```diff
@@ -135,8 +138,29 @@ macro_rules! test_runtest {
         test_declare!($n);
         let res: TestResult = $b;
         match res {
-            TestResult::Pass => test_pass!($n),
-            TestResult::Fail => test_fail!($n),
+            Ok(()) => test_pass!($n),
+            Err(TestFailure::Fail) => test_fail!($n),
+            Err(TestFailure::FailedAt(s)) => test_fail!($n, s),
         }
     };
 }
+
+/// Go from boolean to a TestResult.
+/// True will result in Ok, false will give a FailedAt.
+pub fn test_result_true(expr: bool, name: &'static str) -> TestResult {
```

Comment:
if you keep these as macros, you can use the expression itself as the message for `TestFailure::FailedAt()`, similar to `assert!`.

e.g. `test_result_true!(2 == 3)` becomes `FailedAt("2 == 3")`.